Improve introduction (#143)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

# Introduction

This document describes the scalable, flexible, and decentralized SCITT architecture.
Its goal is to enhance auditability and accountability across supply chains.

> In supply chains, there are producers and consumers of items.

Reviewer: ```suggestion
```

I would remove this line: it adds nothing and is arguably asking for trouble over definition of 'consumer' in particular. Plus what about the stages in the middle? Some people are fine calling these producers and consumers but many aren't.  The next line covers all of this without getting into trouble.

Reviewer 2: If we trim line 132 above, the following 3 lines become orphaned. That may, or may not be a bad thing in the spirit of reduction to create focus, while still maintaining enough context for the goal

Consumers may like to have information about the items that they consume, such as an item's provenance.
SCITT provides a way for consumers to obtain this information in a way that is "transparent", that is, producers cannot lie about the information without it being detected.
SCITT achieves this by having producers (also called Issuers) publish information in a Transparency Service, where consumers (also called Verifiers) can check the information.

SCITT consists of the following roles: Issuers, Transparency Services, Verifiers, and Auditors.
The process works as follows:

> 1. Issuers make Statements about Artifacts they release, for example: "Issuer name", "Artifact type", "Artifact version".

Reviewer 3: In my current mental model, the issuer is the entity that commits to the statement, not necessarily the entity that first made the statement... I don't know if this distinction is worth preserving.

Reviewer 4: That's an interesting point, given that there are many derivative software components from different entities, I think it is important to consider (an Issuer can be the entity who made the software, someone who made a derivative, neither of those and someone who runs conformity/quality testing on software, etc).

Thanks for pointing this out, because I had not internalized this important distinction until now.

Reviewer: Yes, I agree with @OR13 on this one: the only thing we can prove for certain is that the Issuer committed the statement to the TS (and even that's not certain if you want to get all Red Team on us, but we have to draw a line at some point so in the model we assume use of the Issuer Signing Key is foolproof).

As Orie says, they might simply be endorsing a statement made by someone else.
But more importantly it's absolutely not necessarily the case that that Issuer released the Artifact: they might be making observations about someone else's work. 

I'll make a suggestion to this effect

Reviewer 2: agreeing with the points that the issuer is simply "owing" the statement being made about an artifact, not limited to a producer
- Producers of artifacts make statements about their artifacts (Wabbit Networks (issuer) produced some software)
- Auditors make statements about artifacts produced by others ("ScanAuditing" (issuer) attested to an artifact's release process )
- Security companies make statements about the current known quality of artifacts ("CogsWell Security" (issuer) found the net-montior (artifact) shipped by Wabbit Networks is now known to have a vulnerability and no longer meets X123 compliance requirements)
- Consumers make statements about the artifacts they consume, creating a history of their decisions (ACME Rockets (issuer) evaluated the net-monitor software on January 2nd 2024, found it met their security requirements and deployed it)

> 1. Issuers make Statements about Artifacts they release, for example: "Issuer name", "Artifact type", "Artifact version".

Reviewer: ```suggestion
1. Issuers make Statements about Artifacts, for example: "Issuer name", "Artifact type", "Artifact version".
```

Reviewer 2: I would elaborate "Issuer name", "Artifact type", "Artifact version" as these read like placeholders.

> 1. Issuers sign the Statements, producing Signed Statements.

Reviewer 2: When reviewing the document, we often skip to the [architectural overview (visual)](https://ietf-wg-scitt.github.io/draft-ietf-scitt-architecture/draft-ietf-scitt-architecture.html#name-architecture-overview)
Rather than split a textual flow with the visual flow, could these be merged?
Move the visual up to the introduction and provide a textual summary?

Author: Pulled out to #152

1. Issuers submit the Signed Statements to Transparency Services.

> 1. Transparency Services take the submitted Signed Statements and insert them into an Append-only Log, producing Transparent Statements.

Reviewer 2: I'm assuming lines 144-147 are additional text to the numbered bullet on line 143, rather than additional numbered lines? Could the numbered items be summarized more succinctly?

They may add their own metadata such as: "Transparency Service name", "Artifact registration time", "registration policy used".
This process can be thought of as Transparency Services counter-signing Issuers' Signed Statements.
Transparency Services have a Registration Policy controlling what can be submitted.
They may also have an authorization policy controlling who can access their service.

> 1. Verifiers consume the Transparent Statements, verifying them in order to be convinced of the Statement about the Artifact.

Reviewer 3: ```suggestion
1. Verifiers consume the Transparent Statements, verifying them in order to be convinced of the authenticity and integrity of Statements about an Artifact.
```

Reviewer 3: trying to adjust the "order" comment, this currently reads as "verify in [sequential] order", instead of "verify in order to"

Reviewer 2: minor consistency nit: 
- when creating single-sentence bullets, we haven't ended with periods

1. Auditors verify the correctness and consistency of the Transparent Statements.

> An entity can take on multiple roles, for example, an entity can act both as an Issuer and as a Verifier.

Reviewer 2: I wonder if this could be changed to a bulleted list:
```suggestion
An entity can take on multiple roles, for example:

- A Verifier may make new Statements for what they verified
- An Issuer can make multiple Statements about the same Artifact
- An Issuer can make amended Statements about the same Artifact as their view changes over time
- Multiple Issuers can make different, even conflicting Statements, about the same Artifact where another verifier may choose which Issuers they trust
- Multiple Issuers can make the same Statement about a single Artifact, affirming multiple Issuers agree
```

Reviewer 3: This is good.

Author: A few nits:

- Imho the bullets are not examples of "an entity taking on multiple roles at the same time", only the first one is. The others are examples of what a given role can do.
- And even the first bullet is imprecise: the Verifier cannot make Statements, because by design the Issuer does that.  Entity != role.
- The second and third bullet are kind the same / the third is a specification of the second.

Iterated proposal:

```
An entity can take on multiple roles.
For example, an entity can first act as a Verifier, and later act as an Issuer by making a Statement about the verification outcome.

Issuers are free to do any Statements. In particular:

- An Issuer can make multiple Statements about the same Artifact. For example, an Issuer can make amended Statements about the same Artifact as their view changes over time
- Multiple Issuers can make different, even conflicting Statements, about the same Artifact. Verifiers can choose which Issuers they trust
- Multiple Issuers can make the same Statement about a single Artifact, affirming multiple Issuers agree
```

While I like this, I feel like it is getting too much for the introduction again. Can we somehow fit these as clarifying examples into section 2 or  5?

Reviewer 2: The above looks like a cleaner update. 
I'd remove the following line, as the bullets say the same, and the wording of "free to do any" would need updates.

Author: Pulled out to #152

The ability for the original Issuer to make additional Statements about an Artifact provides for updated information to be shared, such as new positive or negative validations of quality.
The ability of other Issuers to make Statements about an Artifact, produced from another Issuer, provides for third party validations.
A Transparency Service may restrict access to Signed Statements through access control or Registration policies.
However, third parties (such as Auditors) would be granted access as needed to attest to the validity of the Artifact, Subject or the entirety of the Transparency Service.
Independent third parties may also make Statements about an Artifact, published on other Transparency Services.

Trust in the Transparency Service itself is supported both by protecting their implementation (using replication, trusted hardware, and remote attestation of a system's operational state) and by enabling independent audits of the correctness and consistency of its Append-only Log, thereby holding the organization that operates it accountable.

> Unlike CT, where independent Auditors are responsible for enforcing the consistency of multiple independent instances of the same global Transparency Service, each Transparency Service is required to guarantee the consistency of its own Append-only Log (through the use of a consensus algorithm between replicas of the Transparency Service), but assume no consistency between different Transparency Services.

Reviewer: Note I have reinstated this line in the 'relation to CT' section

> SCITT is a generalisation of Certificate Transparency {{-CT}}, which can be interpreted as a transparency architecture for the supply chain of X.509 certificates.

Reviewer 3: ```suggestion
## Relation to Certificate Transparency

SCITT is a generalisation of Certificate Transparency {{-CT}}, which can be interpreted as a transparency architecture for the supply chain of X.509 certificates.
```

Reviewer: ```suggestion
SCITT is a generalisation of Certificate Transparency {{-CT}}, which can be interpreted as a transparency architecture for the supply chain of X.509 certificates.

However, the range of use cases and applications in this document is broader, which requires more flexibility in how each Transparency Service is implemented and operates.

Considering CT in terms of SCITT:
```

Reviewer 2: spelling
```suggestion
SCITT is a generalization of Certificate Transparency {{-CT}}, which can be interpreted as a transparency architecture for the supply chain of X.509 certificates.
```

> CAs (Issuers) sign X.509 TBSCertificates (Artifacts) to produce X.509 certificates (Signed Statements).

Reviewer: ```suggestion
 * CAs (Issuers) sign X.509 TBSCertificates (Artifacts) to produce X.509 certificates (Signed Statements).
```

> CAs then submit the certificates to one or more CT logs (Transparency Services).

Reviewer: ```suggestion
 * CAs then submit the certificates to one or more CT logs (Transparency Services).
```

> The CT logs produce Signed Certificate Timestamps (Transparent Statements).

Reviewer: ```suggestion
 * The CT logs produce Signed Certificate Timestamps (Transparent Statements).
```

> The SCTs are checked by browsers (Verifiers), and the Append-only Log can be checked by Auditors.

Reviewer: ```suggestion
* The SCTs are checked by browsers (Verifiers), and the Append-only Log can be checked by Auditors.

Unlike CT, where independent Auditors are responsible for enforcing the consistency of multiple independent instances of the same global Transparency Service, each Transparency Service is required to guarantee the consistency of its own Append-only Log (through the use of a consensus algorithm between replicas of the Transparency Service), but assume no consistency between different Transparency Services.
```

Author: > Unlike CT, where independent Auditors are responsible for enforcing the consistency of multiple independent instances of the same global Transparency Service, each Transparency Service is required to guarantee the consistency of its own Append-only Log (through the use of a consensus algorithm between replicas of the Transparency Service), but assume no consistency between different Transparency Services.

I already asked about this sentence [on the list](https://mailarchive.ietf.org/arch/msg/scitt/1oW4DRFFb1g230VmZVQVgj_Mdw8/) because it doesn't make sense to me.

> multiple independent instances of the same global Transparency Service,

I thought every CT log was a single TS? I.e. there are many independent TS, not "the same global" one.

> each Transparency Service is required to guarantee the consistency of its own Append-only Log

This seems to be true for both CT and SCITT? In both cases we hope that the tree is correctly constructed, but if we want to verify it, the auditors can check the consistency proofs.

> through the use of a consensus algorithm between replicas of the Transparency Service

From [Orie's reply](https://mailarchive.ietf.org/arch/msg/scitt/8IwqR-mw8lKwUHg5YT04CIlGrgQ/), this seems to be an implementation detail of a given TS implementation.

This whole confusing sentence is what prompted me to ask about the CT--SCITT mapping in the first place :)
What are "instances", what are "replicas", what's the difference? Again, see [the list](https://mailarchive.ietf.org/arch/msg/scitt/1oW4DRFFb1g230VmZVQVgj_Mdw8/).

If anything, we could add a sentence that CT consists of many *independent* TS, and that Verifiers may want to query all of them? Or there would need to be a federation mechanism, which isn't fully spec'ed I believe? Or maybe I still understand this wrong?

Reviewer 3: I suggest we remove an reference to concrete deployment models from the architecture.

The transparency service is a single logical entity that maintains a log.

how that is implemented behind the scenes is not relevant to the architecture.

The building blocks defined in SCITT are intended to support applications in any supply chain that produces or relies upon digital artifacts, from the build and supply of software and IoT devices to advanced manufacturing and food supply.

## Relation to Certificate Transparency

SCITT is a generalization of Certificate Transparency {{-CT}}, which can be interpreted as a transparency architecture for the supply chain of X.509 certificates.
Considering CT in terms of SCITT:

> - CAs (Issuers) sign X.509 TBSCertificates (Artifacts) to produce X.509 certificates (Signed Statements)

Reviewer 3: Its my understanding that CT includes modified certificates in the log.

So I might expect the following alignment:

```
TBSCertificates -> Certificate -> Deterministically Broken Cert -> Log Entry
Statement --> Signed Statement -> Log Entry
```

I am not a CT expert, but I would focus on what goes on the log and make it clear that Logs are about "what goes on them".

Author: The tree leaves store the TBSCertificate, among other things:
https://datatracker.ietf.org/doc/html/rfc9162#name-merkle-tree-leaves

Reviewer 2: @OR13, if you're satisfied, can you mark this resolved?
